dist/css-layout.h

```diff
@@ -1064,31 +1064,42 @@ static void layoutNodeImpl(css_node_t* node, float availableWidth, float availab
           }
         }
 
+        // If child has no defined size in the cross axis and is set to stretch, set the cross
+        // axis to be measured exactly with the available inner width
+        if (!isMainAxisRow && !isUndefined(availableInnerWidth) && !isStyleDimDefined(child, CSS_FLEX_DIRECTION_ROW) && widthMeasureMode == CSS_MEASURE_MODE_EXACTLY && getAlignItem(node, child) == CSS_ALIGN_STRETCH) {
```

Comment:
Maybe add a comment to indicate that these branches are for an optimization rather than a functional requirement.